External/HIP/utils/compare_image.py

```diff
@@ -0,0 +1,69 @@
+#!/bin/python3
+import argparse
+import os
+import sys
+
+try:
+    import cv2
+    import numpy as np
+    from skimage.metrics import structural_similarity as ssim
+except ImportError:
+    print("One or more required packages are not installed. Please install them using the command:")
+    print("pip install -r requirements.txt")
+    sys.exit(1)
+
+def mse(imageA, imageB):
+    err = np.sum((imageA.astype("float") - imageB.astype("float")) ** 2)
+    err /= float(imageA.shape[0] * imageA.shape[1] * imageA.shape[2])
+    return err
+
+def compare_images(image_path1, image_path2, ssim_threshold=0.9, mse_threshold=1000):
+    image1 = cv2.imread(image_path1)
+    image2 = cv2.imread(image_path2)
+    
+    if image1 is None or image2 is None:
+        raise ValueError("One or both of the image paths are invalid.")
+    
+    if image1.shape != image2.shape:
+        image2 = cv2.resize(image2, (image1.shape[1], image1.shape[0]))
+    
+    smaller_side = min(image1.shape[:2])
+    win_size = smaller_side if smaller_side % 2 == 1 else smaller_side - 1
+    win_size = max(win_size, 3)
+    
+    ssim_index, diff = ssim(image1, image2, multichannel=True, full=True, win_size=win_size, channel_axis=2)
+    diff = (diff * 255).astype("uint8")
+    
+    mse_value = mse(image1, image2)
+    
+    if ssim_index < ssim_threshold or mse_value > mse_threshold:
```

Comment:
What exactly are we checking for here? 

I assume that the goal is to make sure we produced a sensible image, but it does not have to be pixel-perfect?

I suspect these checks may be rather fragile in general, though it may work well enough for the specific image you use for the test. I guess for a single image we can always find a metric that works, even if blender results change in the future. Still, describing what we do here and why/how the metrics and thresholds were chosen may be helpful.

Reply:
ssim_index is 'structural similarity index' (https://en.wikipedia.org/wiki/Structural_similarity_index_measure), which is a popular measure of image similarity. Since it only measures gray-scale images, mse_value is also used, which measures 'mean square error' (https://en.wikipedia.org/wiki/Mean_squared_error). Together they should be a robust way to spot compiler issues without false alarms.

will add a comment about their meaning.